lib/wpc_proto/internal_modules/proto_config.c

```diff
@@ -391,29 +396,45 @@ app_proto_res_e Proto_config_handle_set_config(wp_SetConfigReq *req,
     // At least refresh stack status
     WPC_get_stack_status(&m_sink_config.StackStatus);
 
-    if (!(m_sink_config.StackStatus & APP_STACK_STOPPED))
+    LOGD("Config received : %d, %d, %d, %d, %d, %d, %d/%d\n",
+            cfg->has_node_role, cfg->has_node_address, cfg->has_network_address,
+            cfg->has_network_channel, cfg->has_keys, cfg->has_current_ac_range,
+            cfg->has_sink_state, cfg->sink_state);
+            
+    if ((m_sink_config.StackStatus & APP_STACK_STOPPED))
+    {
+        // stack already stopped, check if start will be needed
+        if (cfg->has_sink_state && (cfg->sink_state == wp_OnOffState_ON) )
+        {
+            restart_stack = true;
+            config_has_changed = true;
+        }
+    }
+    else
     {
-        // Check if stop is needed
-        if  (    cfg->has_node_role
+        // stack running, check if stop is needed to apply config
+        if  (    (   cfg->has_node_role
+                  && (convert_role_to_app_format(&cfg->node_role) != m_sink_config.app_node_role))
```

Comment:
If you add this check for role, then maybe do it for all other settings. If you set same network_address you will restart.

But now as console is fixed, I think you can remove the test.

Reply:
Moved back.